homework/index.js

```diff
@@ -36,11 +36,115 @@
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        // create the basic element tags //
+        // the main header.
+        createAndAppend('h1', root, { text: 'HYF Repositories' });
+
+        // the container
+        createAndAppend('div', root, { id: 'container' });
+        const container = document.getElementById('container');
+
+        // the basic repository information (left aside).
+        createAndAppend('aside', container, {
+          class: 'left-aside box',
+          id: 'basicInfo',
+        });
+        const basicInfo = document.getElementById('basicInfo');
+
+        // the header of the the basic information repository.
+        createAndAppend('h2', basicInfo, { text: 'Repository Info' });
+
+        // button to hold the name of the selected repository.
+        createAndAppend('button', basicInfo, {
+          id: 'repo-name-btn',
+          value: '',
+        });
+        const repoNameBtn = document.getElementById('repo-name-btn');
+
+        // list of repository basic information.
+        createAndAppend('ul', basicInfo, { id: 'info-ul' });
+        const infoList = document.getElementById('info-ul');
+
+        // select element to select the wanted repository.
+        createAndAppend('select', container, { id: 'select' });
+        const select = document.getElementById('select');
+
+        // the repository contributors (right aside).
+        createAndAppend('aside', container, {
+          class: 'right-aside box',
+          id: 'contributorsAside',
+        });
+        const contributorsAside = document.getElementById('contributorsAside');
+
+        // the header of the contributors aside.
+        createAndAppend('h2', contributorsAside, { text: 'Contributor Info' });
+
+        // ordered list of contributors of each repository selected.
+        createAndAppend('ol', contributorsAside, { id: 'contributors-ol' });
+        const contributorsList = document.getElementById('contributors-ol');
+
+        // sorting the repositories (case-insensitive) on repository name.
+        data.sort((a, b) => a.name.localeCompare(b.name, 'en', { sensitivity: 'base' }));
+        data.forEach(repo => {
+          createAndAppend('option', select, { text: repo.name });
+        });
+
+        // show the first repository basic info on the left aside & the contributors on the right one.
+        const selected = document.getElementById('select');
+
+        const showInfo = function() {
+          data.forEach(repo => {
```

Comment:
Ah - I think there is a better way to do it. You can firstly move the `showInfo` function out of `main` functions scope. Then, this function can accept a repository details as an argument along with some HTML elements in which you would populate the data. This way, you need not have a `forEach` to iterate through all 47 repos returned and having `if-else` check to match it with the currently selected repo.

Thus, your function call for displaying first repo details would look like:
```
showInfo(repository, repoNameBtn, infoList); // add other elements if needed in arguments list
```
and subsequently when this function is called, you will pass the selected element's repo details.

I would suggest that you have a look into the hint provided in the assignment description once: `Add one option element per repository to the select element, where each option element has the array index of the repository as its value attribute and the name of the repository as its text content:`. This would require some changes in lines 88 - 90 of your code and you can easily then move `showInfo` out. Please let me know if you have any questions. 🙂